l10n_ch_invoice_reports/models/account_invoice.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+
+from odoo import api, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    # override to add context key dependency and handle it properly inside method
+    @api.depends_context("invoice_report_no_attachment")
+    def _compute_l10n_ch_isr_valid(self):
+        if self.env.context.get("invoice_report_no_attachment", False):
+            self.update({"l10n_ch_isr_valid": False})
```

Comment:
I think you did not answer one of my prev questions: where this flag needs to be false when printing the report?

Reply:
suppose it should prevent the generation of the attachment in hardcoded implementation on odoo side
https://github.com/odoo/odoo/blob/7fa5965de1c06d2f49e9a2f22a2686d99a19738a/addons/l10n_ch/models/mail_template.py#L36

the comment to method to be updated